master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HAMasterMetaManager.java

```diff
@@ -375,4 +376,23 @@ public void handleUpdatePartitionSize() {
       throw e;
     }
   }
+
+  @Override
+  public void handleApplicationMeta(ApplicationMeta applicationMeta) {
+    try {
+      ratisServer.submitRequest(
+          ResourceRequest.newBuilder()
+              .setCmdType(Type.ApplicationMeta)
+              .setRequestId(MasterClient.genRequestId())
+              .setApplicationMetaRequest(
+                  ResourceProtos.ApplicationMetaRequest.newBuilder()
+                      .setAppId(applicationMeta.getAppId())
+                      .setSecret(applicationMeta.getSecret())
+                      .build())
+              .build());
+    } catch (CelebornRuntimeException e) {
+      LOG.error("Handle app meta failed!", e);
```

Comment:
Do we want to include the app id to help with debugging ?

Reply:
Done